### PR TITLE
Add Flagger

### PR DESCRIPTION
**Pull request by gpaul**
This PR also bumps the istio chart version for recent fixes made to that chart.

- [x] Requires https://github.com/mesosphere/charts/pull/261

### CODE REVIEW

File: templates/istio.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ spec:
   chartReference:
     chart: istio
     repo: https://mesosphere.github.io/charts/staging
-    version: 1.3.3
+    version: 1.3.5
```
Comment (gpaul) [inline templates/istio.yaml]: @GoelDeepak take note, I point kubeaddons-configs to the latest istio chart in mesosphere/charts.

File: templates/flagger.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.19.0"
+    docs.kubeaddons.mesosphere.io/flagger: "https://docs.flagger.app/"
+    values.chart.helm.kubeaddons.mesosphere.io/flagger: "https://raw.githubusercontent.com/weaveworks/flagger/0.19.0/charts/flagger/values.yaml"
+spec:
+  namespace: kubeaddons-flagger
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  cloudProvider:
+    - name: aws
+      enabled: true
```
Comment (GoelDeepak) [inline templates/flagger.yaml]: should we mark it as `false` by default as both `dispatch` and `istio` is disabled by default

**Comment by gpaul** [comment]
@alejandroEsc @shaneutt anything left to be done before we can merge this?

**Comment by gpaul** [comment]
@shaneutt cool, they have been resolved, I believe.

**Comment by gpaul** [comment]
(I don't have commit rights to this repository, so someone else needs to hit merge)

**Pull request merged**